### PR TITLE
feat: common ci workflow

### PR DESCRIPTION
**Pull request by i-chvets**
Implemented common CI as per spec.

### CODE REVIEW

**Comment by i-chvets** [comment]
> Based on the PR name I would expect also a job for publishing the image to the registry. Based on the filename the job is not needed. Not sure we want to add it.
> 
> If we add the publish job we should add an input to CI to specify the version we will publish.

Need to re-hash publishing. It should be manual, because we need to ensure that update image does not introduce more CVEs.

File: .github/workflows/build_and_scan_rock.yaml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+            mkdir -p kubeflow-ci
+            cd kubeflow-ci
+            git init -q
+            git remote add -f origin "$CI_REPO" &> /dev/null
+            git sparse-checkout set scripts/cve-reports/send-scan.py
+            git pull -q origin main
+            cd -
+
+            # send scans from supplied directory
+            ./kubeflow-ci/scripts/cve-reports/send-scan.py --report-path="${{ inputs.rock }}" --jira-url="${{ secrets.JIRA_URL }}"
```
Comment (beliaev-maksim) [inline .github/workflows/build_and_scan_rock.yaml:61]: I think you should list secrets in the beginning of the file

Comment (beliaev-maksim) [reply .github/workflows/build_and_scan_rock.yaml:61]: https://github.com/canonical/charmed-kubeflow-workflows/blob/5e4b2f17b66d7ef72e31ce7923358a26a3ed7fa4/.github/workflows/_publish.yaml#L12

Comment (i-chvets) [reply .github/workflows/build_and_scan_rock.yaml:61]: Strangely enough it worked for our other automation without adding `secrets:` explicitly. I will look into it.

**Review by beliaev-maksim — APPROVED** [review]

**Pull request merged**